### DISCUSSION TITLE
request for wiki or guide or summary of the parameters

### DISCUSSION THREAD

**Discussion by arcolinuxz** [yshui/picom#828] [General]
Just a request to get more information about the possible parameters we can use in the compton.conf.
Would be great if you can reserve one week this year to write down what all the parameters mean.

I have tried to do so here : 

https://github.com/arcolinux/arcolinux-openbox-configs/blob/master/compton/compton.conf

It is based on your compton.conf and then added lines to explain what it does.

Frankly I do not know if it is all correct what I write there.

I have seen some parameters are out and others are completely new.

We could use your guidance and expertise to understand all the codes in the compton.conf.

ArcoLinux is using compton on 

- awesome
- bspwm
- i3
- openbox
- xmonad

Needless to say that compton is important to us and we are glad you guys keep developing it.

As I learn more about compton I certainly will add it to this article.

https://arcolinux.com/all-information-regarding-compton-and-compton-conf/

**Comment by yshui** [comment]
@arcolinuxz Is there anything missing in the man pages?

**Comment by arcolinuxz** [comment]
I am going over the man pages now and I see that what you use in your compton.sample.conf is not entirely the same. 
Other naming conventions, other default settings,

fade-delta = 30; in the compton.sample.conf and in the manual default 10
dropdown_menu = { opacity = 0.8; } in compton.conf and in the manual -m opacity

Would be great if you can add comments to the compton.sample.conf so we are absolutely sure

**Comment by yshui** [comment]
@arcolinuxz So the `compton.sample.conf` is supposed to be a _sample_ configuration file, not the _default_. The default values of options should be the values stated in the man page.

The `dropdown_menu` thing is documented further down the man page.

What I can do is probably just remove all comments in `compton.sample.conf` and point people to the man page. So we won't have two copies of the same information that can potentially go out of sync.

**Comment by arcolinuxz** [comment]
sounds already like an improvement...
Is there a 'default' config that we can use? - I thought this compton.sample.conf could be it.
We are using the compton.conf on 5 different desktops.
Are there recommendations for these 5 desktops?
Are there issues we should be aware of?
Since we use the configuration file - we have other naming conventions from the man page. 
That increases the difficulty to figure out which is which.

**Comment by yshui** [comment]
> Is there a 'default' config that we can use?

We can probably provide a 'default' config, but that is only going to serve a documentation propose, since when you run compton without any config file, it is using the default config. OTOH, we don't want an extra piece of documentation that can go out of sync.

Plus, the 'default' config has options that are likely not what the user wanted (I know, this is bad. This is a problem existed before this fork, and I am still slowly trying to address it).

> Are there recommendations for these 5 desktops?

Which 5 desktops?

I don't think different desktop environments will require specific sets of options for compton.

> Are there issues we should be aware of?

Nothing comes to mind.

> Since we use the configuration file - we have other naming conventions from the man page.

It is easy to go from command line options to config file options (just remove the `--` on the front). I can mention this in the man page.

**Comment by ZirconCode** [comment]
Was not sure where to put this, but what is the correct way to install the man files?
I have asciidoctor installed however the build instructions don't seem to compile the man folder or install it anywhere? `cd man` `asciidoctor -b manpage picom.1.asciidoc` `ls -l picom.1` works.

I was wondering about the deprecated `-d` flag (still mentioned in the archwiki), what would be the current method of starting picom for all displays?

**Comment by Ropid** [comment]
@ZirconCode It seems adding a `-D with_docs=true` to the meson command line makes it compile the man folder.

**Comment by redtide** [comment]
I'm not sure which of this or a new issue is the right place to ask; I'm currently editing the original LXQt compton-conf utility to port it to be used with Picom. It should be nice to have some online documentation to point from some "help" link in the UI, so I'm currently writing a markdown version of the asciidoc documentation on the related project wiki, but it would be nice if there would be an official documentation page.

I thought if we could have a `https://yshui.github.io/picom` page with the auto generated documentation from this repo? I could make a PR, it can be an Actions workflow that builds (using a cron job?) in a "docs" directory here, or in a different repo (say "picom-docs") or in a specified wiki page (I guess it's possible, I never done this before). Thoughts?
